SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -309,51 +335,21 @@ Interest InterestFactory::CreateInterest() const
 		AddPlayerControllerActorInterest(ResultInterest);
 	}
 
-	if (Actor->GetNetConnection() != nullptr && Settings->bEnableResultTypes)
-	{
-		// Clients need to see owner only and server RPC components on entities they have authority over
-		AddClientSelfInterest(ResultInterest);
-	}
-
-	if (Settings->bEnableServerQBI)
-	{
-		// If we have server QBI, every actor needs a query for the server
-		// TODO(jacques): Use worker interest instead (UNR-2656)
-		AddActorInterest(ResultInterest);
-	}
-
 	if (Settings->bEnableResultTypes)
 	{
+		if (Actor->GetNetConnection() != nullptr)
+		{
+			// Clients need to see owner only and server RPC components on entities they have authority over
+			AddClientSelfInterest(ResultInterest);
+		}
+
 		// Every actor needs a self query for the server to the client RPC endpoint
 		AddServerSelfInterest(ResultInterest);
 	}
 
 	return ResultInterest;
 }
 
-void InterestFactory::AddActorInterest(Interest& OutInterest) const
-{
-	QueryConstraint SystemConstraints = CreateSystemDefinedConstraints();
-
-	if (!SystemConstraints.IsValid())
-	{
-		return;
-	}
-
-	Query NewQuery;
-	NewQuery.Constraint = SystemConstraints;
-	if (GetDefault<USpatialGDKSettings>()->bEnableResultTypes)
-	{
-		NewQuery.ResultComponentId = ServerNonAuthInterestResultType;
-	}
-	else
-	{
-		NewQuery.FullSnapshotResult = true;
-	}
-
-	AddComponentQueryPairToInterestComponent(OutInterest, SpatialConstants::POSITION_COMPONENT_ID, NewQuery);
-}
```

Comment:
Have you got others to agree we're fine to delete this functionality off the back of this? I don't quite understand how NCD works entirely so not in a position to really comment. Alternatively, we could put server worker interest behind a flag?

Reply:
entity based server QBI is never turned on, and the changes in this PR make server QBI what people expect it to be. If you are applying per actor interest, you're shooting yourself in the foot, so deleting it.

Reply:
amen